engine/src/flutter/display_list/dl_paint.h

```diff
@@ -123,57 +123,77 @@ class DlPaint {
     return *this;
   }
 
-  std::shared_ptr<const DlColorSource> getColorSource() const {
+  const std::shared_ptr<const DlColorSource>& getColorSource() const {
     return color_source_;
   }
   const DlColorSource* getColorSourcePtr() const { return color_source_.get(); }
-  DlPaint& setColorSource(std::shared_ptr<const DlColorSource> source) {
-    color_source_ = std::move(source);
+
+  DlPaint& setColorSource(std::nullptr_t source) {
```

Comment:
Add nullptr_t overloads. Overload resolution for passing nullptr literal is ambiguous between selecting the raw ptr or shared_ptr. If the shared_ptr path is selected we will construct and destroy a shared_ptr.